diff --git a/gobject/meson.build b/gobject/meson.build
index 78b732bf3..06edbf2f6 100755
--- a/gobject/meson.build
+++ b/gobject/meson.build
@@ -170,10 +170,12 @@ libgobject_dep = declare_dependency(link_with : libgobject,
 )
 meson.override_dependency('gobject-2.0', libgobject_dep)
 
-gobject_query = executable('gobject-query', 'gobject-query.c',
-  install : true,
-  install_tag : 'bin-devel',
-  dependencies : [libglib_dep, libgobject_dep])
+if host_system != 'emscripten'
+  gobject_query = executable('gobject-query', 'gobject-query.c',
+    install : true,
+    install_tag : 'bin-devel',
+    dependencies : [libglib_dep, libgobject_dep])
+endif
 
 install_data('gobject_gdb.py',
   install_dir : glib_pkgdatadir / 'gdb',
